classes/migration/upgrade/v3_5_0/I10620_EditorialBoardMemberRole.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_5_0/I10620_EditorialBoardMemberRole.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I10620_EditorialBoardMemberRole
+ *
+ * @brief Add new Editorial Board Member user group.
+ */
+
+namespace PKP\migration\upgrade\v3_5_0;
+
+use APP\facades\Repo;
+use Illuminate\Support\Facades\DB;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+abstract class I10620_EditorialBoardMemberRole extends Migration
+{
+    abstract protected function getContextTable(): string;
+    abstract protected function getContextSettingsTable(): string;
+    abstract protected function getContextIdColumn(): string;
+
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        $roleId = hexdec('0x00001001');
+        $nameKey = 'default.groups.name.editorialBoardMember';
+        $abbrevKey = 'default.groups.abbrev.editorialBoardMember';
+
+        $installedLocales = json_decode(DB::table('site')->select('installed_locales')->first()->installed_locales, true);
+
+        $contextIds = DB::table($this->getContextTable())
+            ->pluck($this->getContextIdColumn());
+
+        foreach ($contextIds as $contextId) {
+            DB::table('user_groups')->insert([
+                'context_id' => $contextId,
+                'role_id' => $roleId,
+                'is_default' => true,
+                'show_title' => true,
+                'permit_self_registration' => false,
+                'permit_metadata_edit' => false,
+                'permit_settings' => false,
+                'masthead' => true,
+            ]);
+            $userGroupId = (int) DB::getPdo()->lastInsertId();
+
+            DB::table('user_group_settings')->insert([
+                ['user_group_id' => $userGroupId, 'setting_name' => 'nameLocaleKey', 'setting_value' => $nameKey],
+                ['user_group_id' => $userGroupId, 'setting_name' => 'abbrevLocaleKey', 'setting_value' => $abbrevKey],
+            ]);
+
+            foreach ($installedLocales as $locale) {
```

Comment:
This is going to result in a lot of users who move to 3.5.0 immediately (before all of our translations are 100% complete) getting untranslated locale keys installed into the database. I'd rather untranslated names were just not provided in any but the required (primary) locale; that way they'll fall back properly rather than showing `##some.untranslated.key##`.

Reply:
I considered this here, in the upgrade script, but it will be then different when someone installs the new 3.5 version anew. But this is maybe OK so, or? -- Those strings are editable, so if ## appears, the journal manager will eventually easier noticed that it needs to be edited.